server/controllers/file.controller.js

```diff
@@ -7,64 +7,82 @@ import { deleteObjectsFromS3, getObjectKey } from './aws.controller';
 // Bug -> timestamps don't get created, but it seems like this will
 // be fixed in mongoose soon
 // https://github.com/Automattic/mongoose/issues/4049
-export function createFile(req, res) {
-  Project.findOneAndUpdate(
-    {
-      _id: req.params.project_id,
-      user: req.user._id
-    },
-    {
-      $push: {
-        files: req.body
-      }
-    },
-    {
-      new: true
-    }, (err, updatedProject) => {
-      if (err || !updatedProject) {
-        console.log(err);
-        res.status(403).send({ success: false, message: 'Project does not exist, or user does not match owner.' });
-        return;
-      }
-      const newFile = updatedProject.files[updatedProject.files.length - 1];
-      updatedProject.files.id(req.body.parentId).children.push(newFile.id);
-      updatedProject.save((innerErr) => {
-        if (innerErr) {
-          console.log(innerErr);
-          res.json({ success: false });
-          return;
+export async function createFile(req, res) {
+  try {
```

Comment:
same issue here. i don't like the giant `try` body.

Reply:
Makes sense , I will refactor